compiler/rustc_parse/src/parser/path.rs

```diff
@@ -414,17 +413,20 @@ impl<'a> Parser<'a> {
 
     /// Parses a single argument in the angle arguments `<...>` of a path segment.
     fn parse_angle_arg(&mut self) -> PResult<'a, Option<AngleBracketedArg>> {
-        if self.check_ident() && self.look_ahead(1, |t| matches!(t.kind, token::Eq | token::Colon))
-        {
-            // Parse associated type constraint.
-            let lo = self.token.span;
-            let ident = self.parse_ident()?;
-            let kind = if self.eat(&token::Eq) {
-                let ty = self.parse_assoc_equality_term(ident, self.prev_token.span)?;
-                AssocTyConstraintKind::Equality { ty }
-            } else if self.eat(&token::Colon) {
+        let lo = self.token.span;
+        let arg = self.parse_generic_arg()?;
```

Comment:
Ok, I see how https://github.com/rust-lang/rust/pull/79266#discussion_r528208919 works now.

Could you immediately return `Ok(None)` here if `parse_generic_arg` returns `Ok(None)`, and pass `GenericArg` instead of `Option<GenericArg>` to `get_ident_from_generic_arg`?
That should be less confusing.

(And also add a test case for `< = A>` or `< : A>`.)